src/field_theory/finite.lean

```diff
@@ -258,3 +266,17 @@ begin
   simpa only [-zmod.pow_totient, nat.succ_eq_add_one, nat.cast_pow, units.coe_one,
     nat.cast_one, cast_unit_of_coprime, units.coe_pow],
 end
+
+/-- A variation on Fermat's little theorem. See `zmod.fermat_little` -/
+@[simp] lemma zmod.pow_card {p : ℕ} [fact p.prime] (x : zmod p) : x ^ p = x :=
+by { have h := finite_field.pow_card x, rw zmod.card p at h, rw h }
+
+open polynomial
+
+lemma zmod.expand_p {p : ℕ} [fact p.prime] (f : polynomial (zmod p)) :
```

Comment:
Would it be hard to prove this for any field of characteristic `p`?

Reply:
It's not quite true for characteristic p. You have to apply the Frobenius to all the coefficients, or expand by the cardinality of the finite field. I should look at how difficult it is to actually adapt all these proofs today, but some might want to wait for future finite field PRs.

Reply:
I've proved a version for characteristic p, and a version for finite fields in particular.